2 dz/4/main.cpp

```diff
@@ -0,0 +1,99 @@
+﻿/*Напечатать в порядке возрастания все простые несократимые дроби, заключенные между 0 и 1, знаменатели которых не превышают n.*/
+
+#include <iostream>
+#include <iomanip>
+
+using namespace std;
+
+struct Fraction
+{
+    double numerator;
+    double denominator;
+    double arrvalue;
+};
+Fraction *Fr = new Fraction[1000];
+
+int gcd(int a, int b)
+{
+   int c = 0;
+   while (b)
+   {
+      c = a % b;
+      a = b;
+      b = c;
+   }
+   return abs(a);
+}
+
+void quicksort(double *row, int first, int last) //функция сортировки
+{
+    if (first >= last)
+    {
+        return;
+    }
+    double mid = 0;
+    int f = first;
+    int l = last;
+    mid = row[(f + l) / 2];
+    while (f < l)
+    {
+        while (row[f] < mid)
+        {
+            f++;
+        }
+        while (row[l] > mid)
+        {
+            l--;
+        }
+        if (f <= l) //перестановка элементов
+        {
+            swap(row[f], row[l]);
+            f++;
+            l--;
+        }
+    }
+    quicksort(row, first, l);
+    quicksort(row, f, last);
+}
+
+void val(int denominator)
+{
+    double *value = new double[1000];
+    int index = -1;
```

Comment:
It isn't necessary, but will it be better if the `index` will be the current index to insert?
So line 70 -> 75.